### PR TITLE
beater: watch output config changes

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Watch libbeat output config changes, and restart the beater with updated config. This enables us to propagate the Elasticsearch output config to tail-based sampling when running under Fleet.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)~ (non user facing change)
~- [ ] Documentation has been updated~

## How to test these changes

Can't be manually tested yet, needs UI for configuring tail-sampling policies.

## Related issues

#5490

### CODE REVIEW

File: beater/beater.go [hunk 1]
```diff
@@ -845,3 +913,55 @@ func newDropLogsBeatProcessor() (beat.ProcessorList, error) {
...
+			select {
+			case <-r.ctx.Done():
+				return r.ctx.Err()
+			case <-ctx.Done():
+				return ctx.Err()
+			case req.result <- err:
+			}
+		}
+	}
+}
```
Comment (simitt) [inline beater/beater.go:967]: I find the `serve` and `Reload` functionality hard to follow; `Reload` is only called after receiving from the channel, but only `Reload` is sending to the channel. 
Would you mind adding some comment on this `chanReloader`?

Comment (axw) [reply beater/beater.go:976]: I've added some comments, please let me know if they help explain what's going on.

**Review by simitt — APPROVED** [review]
LGTM; thanks for adding the comments.
